app/controllers/application_controller.rb

```diff
@@ -5,6 +5,155 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "  password_security"
   end
 
+  get '/' do 
+    erb :index
+  end
+
+  get '/signup' do
+    #binding.pry
+    if is_logged_in?
+      redirect to '/tweets'
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do 
+    if params.values.include?("")
+      redirect to '/signup'
+    else
+      @user = User.create(params)
+      session[:id] = @user.id
+      redirect '/tweets'
+    end 
+  end
+
+  get '/login' do
+    if is_logged_in?   
+      redirect to '/tweets'
+    else
+       erb :'users/login'
+    end
+  end
+
+  get '/tweets' do 
+    @tweets = Tweet.all
```

Comment:
Wait to set `@tweets` until after validating log in to save an unnecessary call to the database.